api/http/handler/users/user_delete.go

```diff
@@ -94,5 +94,17 @@ func (handler *Handler) deleteUser(w http.ResponseWriter, user *portainer.User)
 		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to remove user memberships from the database", err}
 	}
 
+	// Remove all of the users persisted API keys
+	apiKeys, err := handler.apiKeyService.GetAPIKeys(user.ID)
+	if err != nil {
+		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to retrieve user API keys from the database", err}
+	}
+	for _, k := range apiKeys {
+		err = handler.apiKeyService.DeleteAPIKey(k.UserID, k.ID)
```

Comment:
if we'd keep the service/db logic the way it is, then for a user delete request:
- we fetch all API keys and filter only users'
- for each key we fetch all API keys and filter only a given key
That would be a very expensive call

Reply:
yes i agree, i'll look at changing the underlying logic for the service functions